packages/api/lib/writeRecords/write-granules.js

```diff
@@ -205,17 +205,23 @@ const _writePostgresGranuleViaTransaction = async ({
     granuleRecord,
   });
 
+  // TODO: even if pass in same record, upsertQueryResult will write and return the record.
+  // Should we have a test like upsertQueryResult[0] === pgGranule
```

Comment:
Hmm what would this check? If the upsert result is the same as the granule after the upsert? Meaning if we get an empty `upsertQueryResult` it won't equal `pgGranule` and we'll return the failed status? I think that's what we're doing so I think I'm missing something.